### PR TITLE
hide stream deck settings on mobile

### PR DESCRIPTION
**Pull request by fcannizzaro**

### CODE REVIEW

File: src/app/settings/SettingsPage.tsx [hunk 1]
```diff
@@ -40,6 +40,8 @@ import SortOrderEditor, { SortProperty } from './SortOrderEditor';
 import Spreadsheets from './Spreadsheets';
 import { TroubleshootingSettings } from './Troubleshooting';
 
+const isMobile = 'ontouchstart' in document.documentElement;
```
Comment (bhollis) [inline src/app/settings/SettingsPage.tsx:43]: This isn't a super reliable wY to detect mobile. We usually use isPhonePortrait which detects the mobile layout

Comment (fcannizzaro) [reply src/app/settings/SettingsPage.tsx:43]: I saw that but it works also on landscape orientation ?

Comment (bhollis) [reply src/app/settings/SettingsPage.tsx:43]: Yeah - if your intention is to detect places a streamdeck host couldn't run you'll have to do a lot more. 

Touchstart can be defined on touch-capable laptops/convertible tablets.

Comment (fcannizzaro) [reply src/app/settings/SettingsPage.tsx]: I moved to `matchMedia` using the phone portrait media query

File: src/app/settings/SettingsPage.tsx [hunk 2]
```diff
@@ -544,7 +549,7 @@ export default function SettingsPage() {
 
           <Spreadsheets />
 
-          {$featureFlags.elgatoStreamDeck && <StreamDeckSettings />}
+          {$featureFlags.elgatoStreamDeck && !isMobile && <StreamDeckSettings />}
```
Comment (bhollis) [inline src/app/settings/SettingsPage.tsx:552]:
```suggestion
          {$featureFlags.elgatoStreamDeck && !isPhonePortrait && <StreamDeckSettings />}
```

File: src/app/settings/SettingsPage.tsx [hunk 3]
```diff
@@ -237,7 +240,9 @@ export default function SettingsPage() {
     $featureFlags.wishLists ? { id: 'wishlist', title: t('WishListRoll.Header') } : undefined,
     { id: 'storage', title: t('Storage.MenuTitle') },
     { id: 'spreadsheets', title: t('Settings.Data') },
-    $featureFlags.elgatoStreamDeck ? { id: 'stream-deck', title: 'Elgato Stream Deck' } : undefined,
+    $featureFlags.elgatoStreamDeck && !isMobile
```
Comment (bhollis) [inline src/app/settings/SettingsPage.tsx:243]:
```suggestion
    $featureFlags.elgatoStreamDeck && !isPhonePortrait
```

File: src/app/settings/SettingsPage.tsx [hunk 4]
```diff
@@ -40,6 +40,9 @@ import SortOrderEditor, { SortProperty } from './SortOrderEditor';
 import Spreadsheets from './Spreadsheets';
 import { TroubleshootingSettings } from './Troubleshooting';
 
+// using mobile portrait media query
+const isMobile = window.matchMedia('(max-width: 540px)').matches;
+
```
Comment (bhollis) [inline src/app/settings/SettingsPage.tsx:45]:
```suggestion
```

**Comment by fcannizzaro** [comment]
I don't know how I didn't see the `isPhonePortrait` that is already in this component..

**Review by bhollis — APPROVED** [review]

**Pull request merged**